pandas/core/groupby/ops.py

```diff
@@ -899,10 +899,10 @@ def codes_info(self) -> npt.NDArray[np.intp]:
         return ids
 
     @final
-    def _get_compressed_codes(
+    def _factorize(
         self,
     ) -> tuple[npt.NDArray[np.signedinteger], npt.NDArray[np.intp]]:
-        # The first returned ndarray may have any signed integer dtype
+        """Analogous to core.algorithms.factorize"""
```

Comment:
not sure this is accurate; the second item returned is ndarray[intp] instead of ArrayLike

Reply:
I think it still is the same pattern where `uniques.take[codes]` is the same as the original data, just with a slightly different type. Or is there some bigger difference I'm missing? Should I just change the docstring to be "Similar idea as core.algorithms.factorize", or you don't think algorithms.factorize should be mentioned at all?

Reply:
>  or you don't think algorithms.factorize should be mentioned at all?

yah lets leave this case out